src/Account.sol

```diff
@@ -441,287 +407,335 @@ contract Account is IAccount, OpsReady, Owned, Initializable {
     }
 
-    function _perpsV2ClosePosition(address _market, uint256 _priceImpactDelta)
-        internal
-    {
+    function _perpsV2ClosePosition(address _market, uint256 _priceImpactDelta) internal {
         // establish position
         bytes32 marketKey = IPerpsV2MarketConsolidated(_market).marketKey();
 
         // close position (i.e. reduce size to zero)
         /// @dev this does not remove margin from market
         IPerpsV2MarketConsolidated(_market).closePositionWithTracking(
-            _priceImpactDelta,
-            TRACKING_CODE
+            _priceImpactDelta, TRACKING_CODE
         );
 
         // impose fee (comes from account's margin)
         /// @dev this fee is based on the position's size delta
         _imposeFee(
-            calculateTradeFee({
+            _calculateTradeFee({
                 _sizeDelta: getPosition(marketKey).size,
                 _market: IPerpsV2MarketConsolidated(_market),
-                _advancedOrderFee: 0
+                _conditionalOrderFee: 0
             })
         );
     }
 
     /*//////////////////////////////////////////////////////////////
-                            ADVANCED ORDERS
+                           CONDITIONAL ORDERS
     //////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IAccount
-    function validOrder(uint256 _orderId)
-        public
-        view
+    function placeConditionalOrder(
+        bytes32 _marketKey,
+        int256 _marginDelta,
+        int256 _sizeDelta,
+        uint256 _targetPrice,
+        ConditionalOrderTypes _conditionalOrderType,
+        uint128 _priceImpactDelta,
+        bool _reduceOnly
+    )
+        external
+        payable
         override
-        returns (bool, uint256)
+        notZero(_abs(_sizeDelta), "_sizeDelta")
+        onlyOwner
+        returns (uint256)
     {
-        // Order memory order = getOrder(_orderId);
-
-        // if (order.maxDynamicFee != 0) {
-        //     (uint256 dynamicFee, bool tooVolatile) = exchanger()
-        //         .dynamicFeeRateForExchange(
-        //             SUSD,
-        //             getPerpsV2Market(order.marketKey).baseAsset()
-        //         );
-        //     if (tooVolatile || dynamicFee > order.maxDynamicFee) {
-        //         return (false, 0);
-        //     }
-        // }
-
-        // if (order.orderType == OrderTypes.LIMIT) {
-        //     return validLimitOrder(order);
-        // } else if (order.orderType == OrderTypes.STOP) {
-        //     return validStopOrder(order);
-        // }
-
-        // // unknown order type
-        // // @notice execution should never reach here
-        // // @dev needed to satisfy types
-        return (false, 0);
-    }
+        // ensure account has enough eth to eventually pay for the conditional order
+        if (address(this).balance < MIN_ETH) {
+            revert InsufficientEthBalance(address(this).balance, MIN_ETH);
+        }
```

Comment:
Thinking we maybe get rid of this? Weak assumption because you could end up with multiple orders and still run into issues because you've exhausted all the ETH.

Maybe even just a check that ETH > 0.

Reply:
I was thinking ab this earlier. Currently, users can withdraw ETH even after placing a conditional order. so this check is worthless as is. (could deposit eth, place order, then withdraw it). 

I wonder if we don't restrict eth balance then should we restrict the margin balance?